rosetta-source/src/main/java/cdm/base/staticdata/codelist/ValidateFpMLCodingSchemeImpl.java

```diff
@@ -0,0 +1,155 @@
+package cdm.base.staticdata.codelist;
+
+import cdm.base.staticdata.codelist.functions.LoadCodeList;
+import cdm.base.staticdata.codelist.functions.ValidateFpMLCodingSchemeDomain;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.regnosys.rosetta.common.serialisation.RosettaObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.jar.JarFile;
+import java.util.jar.JarEntry;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Implementation of FpML coding scheme validation.
+ *
+ * This class is responsible for:
+ * - Loading CodeList JSON files from resources using LoadCodeList.
+ * - Caching loaded CodeLists for efficient validation.
+ * - Evaluating whether a given code exists in the relevant CodeList.
+ */
+public class ValidateFpMLCodingSchemeImpl extends ValidateFpMLCodingSchemeDomain {
+
+    private static final Logger logger = LoggerFactory.getLogger(ValidateFpMLCodingSchemeImpl.class);
+
+    /** Directory path where JSON CodeLists are stored inside the resources. */
+    private static final String CODELIST_COLLECTION_PATH = "org/isda/codelist/json";
+
+    private static final ObjectMapper mapper = RosettaObjectMapper.getNewRosettaObjectMapper();
+
+    /** Cache to store loaded CodeLists to avoid redundant file reads. */
+    private static final Map<String, CodeList> cache = new HashMap<>();
```

Comment:
I wonder whether the cache would be better placed inside the `LoadCodeListImpl` class instead? Right now you will actually read the code list multiple files when calling that function.